### PR TITLE
enh(legacy): move login logic into index.php

### PR DESCRIPTION
**Pull request by jeremyjaouen**
## Description

As login and installation doesn't go through index.php with new login mechanism, we can clean the index.php and www/login files. Some part keep maintained to ensure autologin to work or avoid side effects

**Fixes** #MON-12353

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.10.x
- [ ] 21.04.x
- [ ] 21.10.x
- [x] 22.04.x (master)

## Checklist

- [x] I have followed the **coding style guidelines** provided by Centreon
- [] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Review by a-launois — APPROVED** [review]

File: www/index.php [hunk 1]
```diff
@@ -74,82 +57,167 @@
...
+            && $generalOptions["enable_autologin"])
+        || (
+            !isset($generalOptions['sso_enable'])
+            || $generalOptions['sso_enable'] == 1)
+        || (
+            !isset($generalOptions['openid_connect_enable']) || $generalOptions['openid_connect_enable'] == 1)
+    ) {
+        if (isset($_POST['p'])) {
+            $_GET["p"] = $_POST["p"];
+        }
```
Comment (kduret) [inline www/index.php:97]:
```suggestion
        $argP = $_POST['p'] ?? $_GET["p"] ?? null;
```

File: www/index.php [hunk 2]
```diff
@@ -74,82 +57,167 @@
...
+
+        if (isset($_POST['min'])) {
+            $_GET["min"] = $_POST["min"];
+        }
```
Comment (kduret) [inline www/index.php:101]:
```suggestion
        $argMin = $_POST['min'] ?? $_GET["min"] ?? null;
```

File: www/index.php [hunk 3]
```diff
@@ -74,82 +57,167 @@
...
+            && $generalOptions["enable_autologin"])
+        || (
+            isset($_POST["autologin"]) && $_POST["autologin"]
+            && isset($generalOptions["enable_autologin"])
+            && $generalOptions["enable_autologin"])
+        || (
+            !isset($generalOptions['sso_enable'])
+            || $generalOptions['sso_enable'] == 1)
+        || (
+            !isset($generalOptions['openid_connect_enable']) || $generalOptions['openid_connect_enable'] == 1)
```
Comment (kduret) [inline www/index.php:93]: just to be consistent with previous condition
```suggestion
            !isset($generalOptions['openid_connect_enable'])
            || $generalOptions['openid_connect_enable'] == 1)
```

File: www/index.php [hunk 4]
```diff
@@ -74,82 +57,167 @@
...
+        * Check first for Autologin or Get Authentication
+        */
+        $autologin = $_GET["autologin"] ?? CentreonAuthSSO::AUTOLOGIN_DISABLE;
+        $useralias = $_GET["useralias"] ?? null;
+        $password = $passwordG ?? null;
+
+        $token = "";
+        if (isset($_REQUEST['token']) && $_REQUEST['token']) {
+            $token = $_REQUEST['token'];
+        }
```
Comment (kduret) [inline www/index.php:117]:
```suggestion
        $token = $_REQUEST['token'] ?? '';
```

File: www/index.php [hunk 5]
```diff
@@ -74,82 +57,167 @@
...
+
+        $centreonAuth = new CentreonAuthSSO(
+            $dependencyInjector,
+            $useralias,
+            $password,
+            $autologin,
+            $pearDB,
+            $centreonLog,
+            $encryptType,
```
Comment (kduret) [inline www/index.php:130]:
```suggestion
            CentreonAuthSSO::ENCRYPT_MD5,
```

File: www/index.php [hunk 6]
```diff
@@ -74,82 +57,167 @@
...
+
+        $token = "";
+        if (isset($_REQUEST['token']) && $_REQUEST['token']) {
+            $token = $_REQUEST['token'];
+        }
+
+        if (!isset($encryptType)) {
+            $encryptType = CentreonAuthSSO::ENCRYPT_MD5;
+        }
+
```
Comment (kduret) [inline www/index.php:122]:
```suggestion
```

File: www/index.php [hunk 7]
```diff
@@ -74,82 +57,167 @@
...
+                ':providerConfigurationId',
+                $configurationId,
+                \PDO::PARAM_INT
+            );
+            $securityAuthenticationTokenStatement->bindValue(':userId', $centreon->user->user_id, \PDO::PARAM_INT);
+            $securityAuthenticationTokenStatement->execute();
+            $headerRedirection = "./main.php";
+            if (!empty($_GET["p"])) {
+                $headerRedirection .= "?p=" . $_GET["p"];
+                unset($_GET["p"]);
```
Comment (kduret) [inline www/index.php:194]:
```suggestion
            if (!empty($argP)) {
                $headerRedirection .= "?p=" . $argP;
```

File: www/index.php [hunk 8]
```diff
@@ -74,82 +57,167 @@
...
+                );
+                $pearDB->execute($stmt, [$centreon->user->default_page]);
+
+                if ($stmt->rowCount() && ($topologyData = $stmt->fetch()) && $topologyData['is_react']) {
+                    // redirect to the react path
+                    $headerRedirection = '.' . $topologyData['topology_url'];
+                } else {
+                    $headerRedirection .= "?p=" . $centreon->user->default_page;
+
+                    if (isset($_GET["min"]) && $_GET["min"] == '1') {
```
Comment (kduret) [inline www/index.php:213]:
```suggestion
                    if ($argMin == '1') {
```

**Review by kduret — APPROVED** [review]

**Pull request merged**